src/otx/core/model/entity/base.py

```diff
@@ -170,41 +173,67 @@ def map_class_names(src_classes: list[str], dst_classes: list[str]) -> list[int]
                 src2dst.append(-1)
         return src2dst
 
-    def export(self, output_dir: Path, export_format: OTXExportFormat) -> None:
+    def export(
+        self,
+        output_dir: Path,
+        export_format: OTXExportFormatType,
+        precision: OTXExportPrecisionType = OTXExportPrecisionType.FP32,
+    ) -> Path:
         """Export this model to the specified output directory.
 
         Args:
-            output_dir: Directory path to save exported binary files.
-            export_format: Format in which this `OTXModel` is exported.
+            output_dir (Path): directory for saving the exported model
+            export_format (OTXExportFormatType): format of the output model
+            precision (OTXExportPrecisionType): precision of the output model
+        Returns:
+            Path: path to the exported model.
         """
-        if export_format == OTXExportFormat.OPENVINO:
-            self._export_to_openvino(output_dir)
-        if export_format == OTXExportFormat.ONNX:
-            self._export_to_onnx()
-        if export_format == OTXExportFormat.EXPORTABLE_CODE:
-            self._export_to_exportable_code()
+        exporter = self._create_exporter()
+        metadata = self._generate_model_metadata()
 
-    def _export_to_openvino(self, output_dir: Path) -> None:
-        """Export to OpenVINO Intermediate Representation format.
+        if export_format == OTXExportFormatType.OPENVINO:
+            return exporter.to_openvino(self.model, output_dir, self._EXPORTED_MODEL_BASE_NAME, precision, metadata)
+        if export_format == OTXExportFormatType.ONNX:
+            return exporter.to_onnx(self.model, output_dir, self._EXPORTED_MODEL_BASE_NAME, precision, metadata)
+        if export_format == OTXExportFormatType.EXPORTABLE_CODE:
+            return self._export_to_exportable_code()
 
-        Args:
-            output_dir: Directory path to save exported binary files
-        """
+        msg = f"Unsupported export format: {export_format}"
+        raise ValueError(msg)
+
+    def _create_exporter(
+        self,
+    ) -> OTXModelExporter:
+        """Creates OTXModelExporter object that can export the model."""
         raise NotImplementedError
 
-    def _export_to_onnx(self) -> None:
-        """Export to ONNX format.
+    def _generate_model_metadata(
```

Comment:
I'm not sure why `metadata` exists apart from the export parameters, thus we have two sources (`_configure_export_parameters()` and `_generate_model_metadata()`) just for the one `export` feature. Can we bring them into one?

Reply:
I extracted this to higher level, since every model has to menage it's metadata, and metadata tends to be extended when moving down the hierarchy. `_configure_export_parameters` methods are defined only for mmodels for convenience. The generic way to provide parameters for exporter now is to overload _create_exporter() method. I don't want to bring this low-level and weakly structured stuff higher in the hierarchy.

So abstract `OTX[TaskName]Model` knows only about `_generate_model_metadata` and `_create_exporter`, and then every particular child class manages the other details.
In brief, 
metadata: structured, has a clear hierarchical structure, managed from the top of the hierarchy.
export parameters -> a murky dict of anything, created at low levels if needed